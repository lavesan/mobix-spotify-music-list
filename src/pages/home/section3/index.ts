import SectionThreeComponent from './section3.component';

export {
    SectionThreeComponent,
}