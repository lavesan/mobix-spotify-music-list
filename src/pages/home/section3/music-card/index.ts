import MusicCardComponent from './music-card.component';

export {
    MusicCardComponent,
}