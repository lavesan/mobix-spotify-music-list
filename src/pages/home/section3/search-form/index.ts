import SearchFormComponent from './search-form.component';

export {
    SearchFormComponent,
}