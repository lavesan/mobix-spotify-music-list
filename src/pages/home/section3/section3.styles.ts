import styled from 'styled-components';

export const StyledMusicCardsBox = styled.section`
    display: grid;
    grid-template-columns: 3fr 3fr 3fr 3fr;    
    grid-gap: 30px;
    min-height: 0;
    min-width: 0;
    padding: 10px 40px;
    margin-top: 50px;
    max-height: 350px;
    overflow-y: scroll;
`;