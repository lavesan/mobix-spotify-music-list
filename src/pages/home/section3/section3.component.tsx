import React, { useContext } from 'react';
import { StyledH2 } from '../home.styles';
import { SearchFormComponent } from './search-form';
import { StyledMusicCardsBox } from './section3.styles';
import { MusicCardComponent } from './music-card';
import { useHistory } from 'react-router';

export default () => {
    const history = useHistory();

    return (
        <>
            <StyledH2>TODAS AS MÃšSICAS DO ARTISTA</StyledH2>
            <SearchFormComponent />
            <StyledMusicCardsBox>
                <MusicCardComponent
                    albumNumber={1} 
                    trackNumber={1}
                    musicName={'Starway to Heaven'}
                    popularity={'Alta'}
                    explicit={true}
                    onClick={() => history.push({
                        pathname: '/music-info',
                        search: '?name=Parklife'
                        })} />
                <MusicCardComponent
                    albumNumber={20} 
                    trackNumber={43}
                    musicName={'Starway to Heaven'}
                    popularity={'Alta'}
                    explicit={true}
                    onClick={() => history.push({
                        pathname: '/music-info',
                        search: '?name=Starway to Heaven'
                        })} />
                <MusicCardComponent
                    albumNumber={1} 
                    trackNumber={1}
                    musicName={'Starway to Heaven'}
                    popularity={'Alta'}
                    explicit={true}
                    onClick={() => history.push({
                        pathname: '/music-info',
                        search: '?name=Starway to Heaven'
                        })} />
                <MusicCardComponent
                    albumNumber={1} 
                    trackNumber={1}
                    musicName={'Starway to Heaven'}
                    popularity={'Alta'}
                    explicit={true}
                    onClick={() => history.push({
                        pathname: '/music-info',
                        search: '?name=Starway to Heaven'
                        })} />
                <MusicCardComponent
                    albumNumber={1} 
                    trackNumber={1}
                    musicName={'Starway to Heaven'}
                    popularity={'Alta'}
                    explicit={true}
                    onClick={() => history.push({
                        pathname: '/music-info',
                        search: '?name=Starway to Heaven'
                        })} />
                <MusicCardComponent
                    albumNumber={1} 
                    trackNumber={1}
                    musicName={'Starway to Heaven'}
                    popularity={'Alta'}
                    explicit={true}
                    onClick={() => history.push({
                        pathname: '/music-info',
                        search: '?name=Starway to Heaven'
                        })} />
                <MusicCardComponent
                    albumNumber={1} 
                    trackNumber={1}
                    musicName={'Starway to Heaven'}
                    popularity={'Alta'}
                    explicit={true}
                    onClick={() => history.push({
                        pathname: '/music-info',
                        search: '?name=Starway to Heaven'
                        })} />
            </StyledMusicCardsBox>
        </>
    )
}