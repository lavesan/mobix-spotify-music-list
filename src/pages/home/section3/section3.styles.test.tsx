import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import 'jest-styled-components';
import { StyledMusicCardsBox } from './section3.styles';

describe('Tests `StyledMusicCardsBox` ', () => {
  let tree: ReactTestRendererJSON | null;

  beforeAll(() => {
    tree = renderer.create(<StyledMusicCardsBox />).toJSON();
  })

  test('Matches snapshot ', () => {
    expect(tree).toMatchSnapshot();
  })
  
  test('Have styles ', () => {
    expect(tree).toHaveStyleRule('grid-template-columns', '1fr 1fr 1fr 1fr')
    expect(tree).toHaveStyleRule('grid-template-columns', '1fr 1fr 1fr', {
      media: '(max-width: 1100px)',
    })
    expect(tree).toHaveStyleRule('grid-template-columns', '1fr 1fr', {
      media: '(max-width: 860px)',
    })
    expect(tree).toHaveStyleRule('grid-template-columns', '1fr', {
      media: '(max-width: 600px)',
    })
  })
})
