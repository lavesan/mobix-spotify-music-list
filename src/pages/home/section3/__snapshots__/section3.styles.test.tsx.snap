// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests \`StyledMusicCardsBox\`  Matches snapshot  1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 30px;
  min-height: 0;
  min-width: 0;
  padding: 10px 40px;
  margin-top: 50px;
  max-height: 350px;
  overflow-y: scroll;
}

@media (max-width:1100px) {
  .c0 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media (max-width:860px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
  }
}

@media (max-width:600px) {
  .c0 {
    grid-template-columns: 1fr;
  }
}

<section
  className="c0"
/>
`;
